validation/static/editSegment.js

```diff
@@ -0,0 +1,46 @@
+"use strict";
+
+document.addEventListener('DOMContentLoaded', () => {
+    for (let button of document.querySelectorAll('[data-segment-name]')) {
+        button.addEventListener("click", async (e) => {
+            showDiv()
+            const name = e.target.dataset.segmentName
+            const translation = document.getElementById(name + '-translation').innerHTML;
+
+            document.getElementById('id_cree').value = name;
+            document.getElementById('id_translation').value = translation;
+        })
+    }
+
+    for (let button of document.querySelectorAll('[data-cy="accept-button"], [data-cy="revert-button"]')) {
+        button.addEventListener("click", async(e) => {
+            showDiv()
+
+            const transcription = e.target.dataset.transcription
+            const translation = e.target.dataset.translation
+            const analysis = e.target.dataset.analysis
+
+            document.getElementById('id_cree').value = transcription;
+            document.getElementById('id_translation').value = translation;
+            document.getElementById('id_analysis').value = analysis;
+        })
+    }
+
+    for (let button of document.querySelectorAll('[data-cy="cancel-button"]')) {
+        button.addEventListener("click", async(e) => {
+            document.getElementById('edit').style.display = "none";
+        })
+    }
+
+    for (let button of document.querySelectorAll('[data-cy="back-button"]')) {
+        button.addEventListener("click", async(e) => {
+            window.history.back()
+        })
+    }
+})
+
+
+function showDiv() {
+    document.getElementById('edit').style.display = "block";
+    document.getElementById('id_cree').focus()
```

Comment:
Note: assigning focus programmatically is considered [not a great practice](https://stackoverflow.com/a/2180699/6626414), but here, I don't think it matters much. Always be careful with focus stealing.